src/Magnum/Platform/EmscriptenApplication.h

```diff
@@ -0,0 +1,1509 @@
+
+        /** @copydoc Sdl2Application::drawEvent() */
+        virtual void drawEvent() = 0;
+
+        /*@}*/
+
+        /** @{ @name Keyboard handling */
+
+        /** @copydoc Sdl2Application::keyPressEvent() */
+        virtual void keyPressEvent(KeyEvent& event);
+
+        /** @copydoc Sdl2Application::keyReleaseEvent() */
+        virtual void keyReleaseEvent(KeyEvent& event);
+
+        /*@}*/
+
+        /** @{ @name Mouse handling */
+
+    private:
+        /**
+         * @brief Mouse press event
+         *
+         * Called when mouse button is pressed. Default implementation does
+         * nothing.
+         */
+        virtual void mousePressEvent(MouseEvent& event);
+
+        /**
+         * @brief Mouse release event
+         *
+         * Called when mouse button is released. Default implementation does
+         * nothing.
+         */
+        virtual void mouseReleaseEvent(MouseEvent& event);
+
+        /**
+         * @brief Mouse move event
+         *
+         * Called when mouse is moved. Default implementation does nothing.
+         */
+        virtual void mouseMoveEvent(MouseMoveEvent& event);
+
+        /**
+         * @brief Mouse scroll event
+         *
+         * Called when a scrolling device is used (mouse wheel or scrolling
+         * area on a touchpad). Default implementation does nothing.
+         */
+        virtual void mouseScrollEvent(MouseScrollEvent& event);
+
+        /*@}*/
+
+        /** @{ @name Text input handling */
+    public:
+        /**
+         * @brief Whether text input is active
+         *
+         * If text input is active, text input events go to @ref textInputEvent()
+         * and @ref textEditingEvent().
+         * @note Note that in @ref CORRADE_TARGET_EMSCRIPTEN "Emscripten" the
+         *      value is emulated and might not reflect external events like
+         *      closing on-screen keyboard.
+         * @see @ref startTextInput(), @ref stopTextInput()
+         */
+        bool isTextInputActive() const {
+            return !!(_flags & Flag::TextInputActive);
+        }
+
+        /**
+         * @brief Start text input
+         *
+         * Starts text input that will go to @ref textInputEvent() and
+         * @ref textEditingEvent().
+         * @see @ref stopTextInput(), @ref isTextInputActive(),
+         *      @ref setTextInputRect()
+         */
+        void startTextInput();
+
+        /**
+         * @brief Stop text input
+         *
+         * Stops text input that went to @ref textInputEvent() and
+         * @ref textEditingEvent().
+         * @see @ref startTextInput(), @ref isTextInputActive(), @ref textInputEvent()
+         *      @ref textEditingEvent()
+         */
+        void stopTextInput();
+
+        /**
+         * @brief Set text input rectangle
+         *
+         * The @p rect defines an area where the text is being displayed, for
+         * example to hint the system where to place on-screen keyboard.
```

Comment:
Can you put a "Currently not implemented, included only for compatibility with other Application implementations" here?